pandas/core/arrays/datetimes.py

```diff
@@ -466,22 +466,19 @@ def _sub_datelike_dti(self, other):
         return new_values.view('timedelta64[ns]')
 
     def _add_offset(self, offset):
-        assert not isinstance(offset, Tick)
         try:
             if self.tz is not None:
                 values = self.tz_localize(None)
             else:
                 values = self
             result = offset.apply_index(values)
-            if self.tz is not None:
-                result = result.tz_localize(self.tz)
 
         except NotImplementedError:
             warnings.warn("Non-vectorized DateOffset being applied to Series "
                           "or DatetimeIndex", PerformanceWarning)
             result = self.astype('O') + offset
 
-        return type(self)(result, freq='infer')
+        return type(self)(result, freq='infer', tz=self.tz)
```

Comment:
This was changed because if the code goes down the `NotImplementedError` path, the tz would be dropped.